### PR TITLE
refactor: simplify OrderedDict arguments in lexer

### LINKED ISSUE #599: Simplify passing the arguments to OrderedDict in lexer
[Token definitions](https://github.com/googleapis/python-bigquery/blob/be4961257f077b96b595cfcd6553650bd4c618ad/google/cloud/bigquery/magics/line_arg_parser/lexer.py#L54) in the custom parser for bigquery cell magic are passed as a list of (key, value) pairs, which results in extra two levels of indentation after formatting and additional noise (parentheses, quotes...).

Since we don't support Python 3.5 and below anymore, we can define the tokens as kwargs to OrderedDict. The kwargs ordered in Python 3.6+ is preserved ([PEP 468](https://www.python.org/dev/peps/pep-0468/)).

### PR DESCRIPTION
**Pull request by plamut**
Closes #599. 🦕

Python 3.6+ guarantees that kwargs order is preserved, thus we don't need to assure the order by passing the arguments as a list of tuples (we don't support Python 3.5 anymore). Results in a more compact and cleaner source (IMO).

There are still some `OrderedDict` instances in tests that could benefit from the same, but I'm not touching that to not cause merge conflicts with the tests refactoring changes @jimfulton is making right now.

**PR checklist**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by plamut** [comment]
@jimfulton Assigned to you if you have free cycles and want to get more familiar with the BigQuery code.

**Comment by tswast** [comment]
> Python 3.6+ guarantees that kwargs order is preserved

This is true for CPython, but I recall that the language standard only makes this the case for Python 3.7+. Not that we support/test with alternative runtimes... Maybe worth bringing up in the Python meeting today.

File: google/cloud/bigquery/magics/line_arg_parser/lexer.py [hunk 1]
```diff
@@ -49,90 +49,45 @@
 # the value of an option other than "--params", we do not really care about its
 # structure, and thus do not want to use any of the "Python tokens" for pattern matching.
 #
-# Since token definition order is important, an OrderedDict is needed with tightly
-# controlled member definitions (i.e. passed as a sequence, and *not* via kwargs).
+# Token definition order is important, thus an OrderedDict is used. In addition, PEP 468
+# guarantees us that the order of kwargs is preserved in Python 3.6+.
```
[the remark this answers is not in the document]
Comment (tswast) [reply google/cloud/bigquery/magics/line_arg_parser/lexer.py:53]: Ah! I didn't realize kwargs became ordered a version earlier :-)

**Review by tswast — APPROVED** [review]
Much cleaner!

**Review by jimfulton — APPROVED** [review]

**Pull request merged**